shark_turbine/kernel/wave/visualization.py

```diff
@@ -12,9 +12,9 @@ def visualize_graph(graph: fx.Graph, file_name: str):
         raise ImportError("pygraphviz not installed, cannot visualize graph")
     G = pgv.AGraph(directed=True)
     for node in graph.nodes:
-        G.add_node(node.name)
+        G.add_node(id(node), label=node.name)
```

Comment:
`id(node)` is essentially a pointer, it is not stable between runs, we can potentially get different files each time (even if they look the same).

Reply:
We can iterate over all nodes, number them starting from 0 and, build a map `{id(node) : int}` and use resulting int as stable id.

Feel free to do it in separate PR to get things going fast.

Reply:
Thanks, will do that.